### PR TITLE
refactor: Bump terraform sdk to version 2

### PR DESCRIPTION
**Pull request by nodece**
Fix: #36

This PR was implemented based on #30.

### CODE REVIEW

File: pulsar/provider.go [hunk 1]
```diff
@@ -37,33 +37,36 @@ func Provider() terraform.ResourceProvider {
 		Schema: map[string]*schema.Schema{
 			"web_service_url": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
```
Comment (sijie) [inline pulsar/provider.go:40]: why is this changed to `Optional`?

Comment (nodece) [reply pulsar/provider.go:40]: `terraform-plugin-sdk` doesn't support the set `Required: true` in provider, so I added the default value for `web_service_url` in init the provider, when the `web_service_url` is empty, which will be set to `http://localhost:8080`.

File: hashcode/hashcode.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package hashcode
+
+import (
+	"bytes"
+	"fmt"
+	"hash/crc32"
+)
+
+// Implementation fro https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html
```
Comment (maxsxu) [inline hashcode/hashcode.go:9]: typo: `fro` -> `from`

Comment (maxsxu) [inline hashcode/hashcode.go:9]: typo: `fro` -> `from`

File: hashcode/hashcode.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+// String hashes a string to a unique hashcode.
+//
+// crc32 returns a uint32, but for our use we need
```
Comment (maxsxu) [inline hashcode/hashcode.go:13]: typo: `a uint32` -> `an uint32`

Comment (maxsxu) [inline hashcode/hashcode.go:13]: typo: `a uint32` -> `an uint32`

File: hashcode/hashcode.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+// and non negative integer. Here we cast to an integer
```
Comment (maxsxu) [inline hashcode/hashcode.go:14]: typo: `and non negative` -> `a non-negative`

Comment (maxsxu) [inline hashcode/hashcode.go:14]: typo: `and non negative` -> `a non-negative`

**Comment by ericsyh** [comment]
@nodece Max helped take a reivew on your PR, pls check his comments.

**Comment by ericsyh** [comment]
@jrsdav @jdmaguire Hi Joey, John, Platform eng Zixuan is working on terraform SDK upgrade and wanna release a new version of terraform-provider-pulsar. Could you also help take a look at this PR since Platform team doesn't have much experience on terraform.

**Comment by nodece** [comment]
@maxsxu Thanks for your review, these request has been addressed.

**Review by maxsxu — APPROVED** [review]

**Review by maxsxu — APPROVED** [review]
